src/views/ChartPage.vue

```diff
@@ -0,0 +1,463 @@
+    },
+    drawPieChart () {
+      const labelList = this.getNames(this.masterCategoryId)
+      const dataList = this.getDatas(this.typeInformationPie, this.masterCategoryId)
+      const colorList = this.getColor(false, false, false, dataList.length)
+      this.pieChartData.labels = labelList
+      this.pieChartData.datasets[0].data = dataList
+      this.pieChartData.datasets[0].backgroundColor = colorList
+    },
+    drawBarChart () {
+      this.chartData.datasets = []
+      if (this.typeInformationBar.indexOf('available') !== -1) {
+        const datasetsAvailable = {
+          label: 'Disponible',
+          backgroundColor: this.getColor(false, false, true, 0),
+          data: this.getDatas('available', this.masterCategoryId)
+        }
+        this.chartData.datasets.splice(0, 0, datasetsAvailable)
+      }
+      if (this.typeInformationBar.indexOf('spent') !== -1) {
+        const datasetsSpent = {
+          label: 'Dépensé',
+          backgroundColor: this.getColor(false, true, false, 0),
+          data: this.getDatas('spent', this.masterCategoryId)
+        }
+        this.chartData.datasets.splice(0, 0, datasetsSpent)
+      }
+      if (this.typeInformationBar.indexOf('allocated') !== -1) {
+        const datasetsAllocation = {
+          label: 'Allocation',
+          backgroundColor: this.getColor(true, false, false, 0),
+          data: this.getDatas('allocated', this.masterCategoryId)
+        }
+        this.chartData.datasets.splice(0, 0, datasetsAllocation)
+      }
+      this.chartData.labels = this.getNames(this.masterCategoryId)
+    },
+    changeGraph () {
+      switch (this.currentGraph) {
+        case 'pie':
+          this.currentGraph = 'bar'
+          break
+        default:
+          this.currentGraph = 'pie'
+      }
+    },
+    getMonthAsDate (monthAsInt: number): Date {
+      return Time.getMonthAsDate(monthAsInt)
+    },
+    async goToNextMonth () {
+      this.budgetMonth = Time.getNextMonth(this.budgetMonth)
+      await this.getBudgetData()
+      this.recalculate()
+    },
+    async goToLastMonth () {
+      this.budgetMonth = Time.getLastMonth(this.budgetMonth)
+      await this.getBudgetData()
+      this.recalculate()
+    },
+    recalculate () {
+      this.drawPieChart()
+      this.drawBarChart()
+    },
+    switchTypeInformationPie (newType: string) {
+      switch (newType) {
+        case 'allocated':
+          this.typeInformationPie = 'allocated'
+          this.recalculate()
+          break
+        case 'spent':
+          this.typeInformationPie = 'spent'
+          this.recalculate()
+          break
+        case 'available':
+          this.typeInformationPie = 'available'
+          this.recalculate()
+          break
+        default:
+          this.typeInformationPie = ''
+      }
+    },
+    switchTypeInformationBar (newType: string) {
+      switch (newType) {
+        case 'allocated':
+          if (this.typeInformationBar.indexOf('allocated') === -1) {
```

Comment:
idem the behavior seems to be the same for all part of the case, can you factorize it in a shorter way ?

Reply:
done